File: sonarqube-companion-rest/src/main/java/pl/consdata/ico/sqcompanion/config/model/ServerDefinition.java
package pl.consdata.ico.sqcompanion.config.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Data
@Builder(toBuilder = true)
@NoArgsConstructor
@AllArgsConstructor
public class ServerDefinition {

    private String id;
    private String uuid;
    private String url;
    private ServerAuthentication authentication;
    private List<String> blacklistUsers;
    private List<String> aliases;

    public boolean hasAuthentication() {
        return authentication != null
                && authentication.getType() != null
                && !authentication.getType().isEmpty()
                && !authentication.getType().toLowerCase().equals("none");
    }

    public List<String> getBlacklistUsers() {
        return blacklistUsers != null ? blacklistUsers : new ArrayList<>();
    }

}

File: sonarqube-companion-frontend/src/app/config/model/server-definition.ts
import {AuthenticationData} from './authentication-data';

export class ServerDefinition {
  id: string;
  uuid: string;
  url: string;
  authentication: AuthenticationData;
  blacklistUsers: string[];
  aliases: string[];


  constructor(data?: any) {
    if (data) {
      this.id = data.id;
      this.uuid = data.uuid;
      this.url = data.url;
      this.authentication = new AuthenticationData(data.authentication);
      this.blacklistUsers = data.blacklistUsers;
      this.aliases = data.aliases;
    } else {
      this.authentication = new AuthenticationData();
    }
  }


}
